modules/server/src/main/ts/bedrock/server/Controller.ts

```diff
@@ -107,9 +108,16 @@ export const create = (stickyFirstSession: boolean, singleTimeout: number, overa
     outputToHud = true;
   };
 
+  const shouldUpdateHud = (session: TestSession): boolean => {
+    if (!outputToHud) return false;
+    if (stickyFirstSession && (timeoutError || session.id !== stickyId)) return false;
+    if (!Env.IS_CI || session.done) return true;
+    // Only update the HUD at 10% intervals on remote:
+    return session.results.length % Math.round(session.totalTests * 0.1) === 0;
+  };
+
   const updateHud = (session: TestSession) => {
-    if (!outputToHud) return;
-    if (stickyFirstSession && (timeoutError || session.id !== stickyId)) return;
+    if (!shouldUpdateHud(session)) return;
```

Comment:
I'm torn about whether to keep this on the server or just reduce the frequency of traffic from the client.

Lower traffic should speed up the connection on slow tunnels, but might risk incorrect HUD output when a test times out.

Reply:
Maybe it should always log tests that didn't pass as well?
```typescript
if (!Env.IS_CI || session.done || !session.results.at(-1)?.passed) return true;
```

Reply:
What would be really nice is
* reduce frequency of client to HUD status push to 10% in CI
* when there is a test failure, override this and do a push immediately

Reply:
But this is at least _better_, so I think publish it and if you like work on that idea as a new PR.